### DISCUSSION TITLE
Authelia PUSH error with DUO after update

### DISCUSSION THREAD

**Discussion by tltglitch** [authelia/authelia#6955] [Q&A]
After authelia update to version "authelia version v4.38.4" i  now have issues with users not being able to authenticate with duo. Here is the error in the log message:

Failed to perform Duo PreAuth for user 'openheimer': Post \"https://api-XXXXXXX.duosecurity.com/auth/v2/preauth\": tls: failed to verify certificate: x509: certificate signed by unknown authority" method=POST path=/api/secondfactor/duo remote_ip=20.20.20.20 stack="github.com/authelia/authelia/v4/internal/handlers/response.go:287         respondUnauthorized\ngithub.com/authelia/authelia/v4/internal/handlers/handler_sign_duo.go:148 HandlePreferredDeviceCheck\ngithub.com/authelia/authelia/v4/internal/handlers/handler_sign_duo.go:48  handleRouter.DuoPOST.func15\ngithub.com/authelia/authelia/v4/internal/middlewares/require_auth.go:19   Require1FA.func1\ngithub.com/authelia/authelia/v4/internal/middlewares/bridge.go:54         handleRouter.(*BridgeBuilder).Build.func6.1\ngithub.com/authelia/authelia/v4/internal/middlewares/headers.go:65        SecurityHeadersCSPNone.func1\ngithub.com/authelia/authelia/v4/internal/middlewares/headers.go:91        SecurityHeadersNoStore.func1\ngithub.com/authelia/authelia/v4/internal/middlewares/headers.go:30        SecurityHeadersBase.func1\ngithub.com/fasthttp/router@v1.5.0/router.go:441                           (*Router).Handler\ngithub.com/authelia/authelia/v4/internal/middlewares/log_request.go:14    handleRouter.LogRequest.func40\ngithub.com/authelia/authelia/v4/internal/middlewares/errors.go:38         RecoverPanic.func1\ngithub.com/valyala/fasthttp@v1.52.0/server.go:2374                        (*Server).serveConn\ngithub.com/valyala/fasthttp@v1.52.0/workerpool.go:224                     (*workerPool).workerFunc\ngithub.com/valyala/fasthttp@v1.52.0/workerpool.go:196                     (*workerPool).getCh.func1\nruntime/asm_amd64.s:1695

Our authelia ssl certs are all valid. I checked using an ssl validator.

Not sure what the issue is.  Can anyone help please?

**Comment by glime-ai** [comment]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

The error message you're encountering indicates that Authelia is unable to verify the Duo Security API's SSL certificate. This is typically because the root CA (Certificate Authority) that signed the Duo Security API's certificate is not trusted by the system on which Authelia is running. Even though your Authelia SSL certificates are valid, the issue here is with the trust chain between Authelia and Duo's API.

To resolve this issue, you need to ensure that the root CA certificate that signed Duo's API certificate is included in the trust store of the system or explicitly specified in Authelia's configuration. Authelia allows specifying custom root CA certificates for outbound connections in its configuration. You can do this by adding the root CA certificate that signed Duo's API certificate to Authelia's configuration under the `certificates_directory` or directly specifying it in the relevant section if you're making a direct HTTPS call to Duo's API.

Here are the steps you should follow:

1. **Identify and Obtain the Root CA Certificate**: You need to identify the root CA certificate that signed the certificate used by `api-XXXXXXX.duosecurity.com`. You can do this by visiting the URL in a web browser and viewing the certificate information, or using a tool like OpenSSL to fetch the certificate chain.

2. **Update Authelia's Configuration**: Once you have the root CA certificate, you need to add it to Authelia's configuration. If you're using a version of Authelia that supports specifying root CA certificates for outbound connections, you can add the path to the root CA certificate file in the configuration under the appropriate section (e.g., `notifier` or `authentication_backend`).

3. **Restart Authelia**: After updating the configuration, restart Authelia to apply the changes.

If you're unable to find the specific configuration option or if the issue persists after adding the root CA certificate, I recommend checking Authelia's official documentation or reaching out to the Authelia support community for more detailed assistance based on your specific setup and version.

Please note that the exact configuration option and steps may vary depending on the version of Authelia you're using, so it's important to refer to the documentation for the version you have installed.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).

**Comment by james-d-elliott** [comment]
This indicates the SSL certs at DUO are invalid for that endpoint, since we use the official DUO go library this seems very unusual.

**Reply by tltglitch** [reply]
Thank you for your quick response. Yes it does and i cannot figure out why it is happening. Any pointers ?

**Reply by james-d-elliott** [reply]
What happens if you visit the api endpoint directly?

**Reply by tltglitch** [reply]
@james-d-elliott Thank you for your support. After 24hrs of troubleshooting, and posting here as a last resort,  your last comment and the AI bots response had me looking more and i decided to do a packet capture at the firewall level. Nothing has changed in our network but it now seems after the update of our authelia instance, the duo api does not like "deep packet inspection" the firewall is performing and has been performing for years lol. Creating an exemption for the endpoint URL fixed the issue. So this was not an authelia issue. Thank you again.

**Reply by james-d-elliott** [reply]
Very strange it worked at all honestly lol, that usually indicates a Enterprise PKI which you'd have to trust the certificate for.

**Reply by tltglitch** [reply]
We only DPI for outbound connections so all our clients have the cert installed. The logs shows DUO denying authelia's response due to the dpi cert. Some endpoints or applications do not like or have issues with dpi , putting an exception lets our firewalls know not to dpi for those websites in this case we put one for *duosecurity.com and it fixed the issue. Im not sure why it worked before with no exemption, duo must have changed something on their end.

**Reply by james-d-elliott** [reply]
Yeah could be, or could be a security enhancement in go
